src/main/kotlin/org/axonframework/intellij/ide/plugin/api/MessageHandlerType.kt

```diff
@@ -0,0 +1,72 @@
+package org.axonframework.intellij.ide.plugin.api
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.JavaPsiFacade
+import com.intellij.psi.PsiClass
+import com.intellij.psi.search.GlobalSearchScope.allScope
+import com.intellij.psi.search.searches.AnnotatedElementsSearch
+import com.intellij.psi.util.CachedValue
+import com.intellij.psi.util.CachedValueProvider
+import com.intellij.psi.util.CachedValuesManager
+import com.intellij.psi.util.PsiModificationTracker
+import java.util.concurrent.ConcurrentHashMap
+
+enum class MessageHandlerType(
+        val messageType: MessageType?,
```

Comment:
Sharp! I'll make it non-nullable. I think I made it nullable in the past for more generic handlers (with the `@MessageHandler` annotation), but later opted for more specific ones instead.

Reply:
Aaah alright, that's actually not very off.
When we'd for example introduce a new annotation based on the `MessageHandler`, with a different type of message in it, it might still be supported then. 
Whether that's reasonable is a different topic though.

Reply:
While on the point of message handler annotations: note that users can use custom annotations, meta-annotating them with Axon's annotations.
It would be fantastic if the plugin would keep supporting the gutter icons whenever some end-user utilizes this approach.
By no means something we should get in ASAP, but a nice feature nonetheless.